sailor/sap_iot/write.py

```diff
@@ -68,14 +70,54 @@ def _upload_data_single_indicator_group(dataset, indicator_set, group_id, templa
         _upload_data_single_equipment(data_subset, equipment.id, tags)
 
 
-def upload_indicator_data(dataset: TimeseriesDataset):
+def _check_indicator_group_is_complete(uploaded_indicators, group_id, template_id):
+    missing = []
+    ig_id = group_id.replace('IG_', '')
+
+    request_url = _ac_application_url() + VIEW_TEMPLATES + '/' + template_id
+    oauth_ac = get_oauth_client('asset_central')
+    template = oauth_ac.request('GET', request_url)
+    for item in template:
+        for ig in (filter(lambda x: x['id'] == ig_id, item['indicatorGroups'])):
+            group_name = ig['internalId']
+
+            for indicator in (filter(lambda x: x['internalId'] not in uploaded_indicators, ig['indicators'])):
+                missing.append(indicator['internalId'])
```

Comment:
similar here, why the for loop? something like `missing.extend(filter(...))` should be all you need, no?